### PR TITLE
Feature/bank structure

### PR DESCRIPTION
**Pull request by benkolera**
Splits the server up a little and gets a non-handler context in there.

I'll keep this branch open and do more work and PRs after: 
- split into Common.Customer, Banking.Transactions, Banking.Payees, Banking.Products
- A model layer that we load the fake data from a conf or something
- Getting some kind of auth or use selection so that the model can choose to return a person, organisation or 404 for a common.customer call, for example

### CODE REVIEW

File: consumer-data-au-lambdabank/src/Web/ConsumerData/Au/LambdaBank/Server.hs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  , _banking = bankingServer
+  }
+
+app :: LinkQualifier -> Application
+app lq = genericServeT (runLambdaBankM lq) routes
+  where
+    runLambdaBankM :: LinkQualifier -> LambdaBankM a -> Handler a
+    runLambdaBankM lq m = pure $ runReader m lq
+
+-- | (Remove this when we go to servant 0.15)
```
Comment (benkolera) [inline consumer-data-au-lambdabank/src/Web/ConsumerData/Au/LambdaBank/Server.hs]: It's probably just better to borrow this for now given it is bsd code and nixpkgs should catch up soon.

**Comment by benkolera** [comment]
Dang I love how easy it is to wire a reader through there! :heart:

File: consumer-data-au-lambdabank/src/Web/ConsumerData/Au/LambdaBank/Server.hs (hunk 1 above)
Comment (gwils) [reply consumer-data-au-lambdabank/src/Web/ConsumerData/Au/LambdaBank/Server.hs]: Do it

Comment (benkolera) [reply consumer-data-au-lambdabank/src/Web/ConsumerData/Au/LambdaBank/Server.hs]: Do it, as it fine to borrow till nixpkgs catches up? It's a pretty simple, but handy little thing.

Comment (gwils) [reply consumer-data-au-lambdabank/src/Web/ConsumerData/Au/LambdaBank/Server.hs]: Yeah, fine to borrow. We might need to put a copyright disclaimer somewhere though.

File: consumer-data-au-lambdabank/consumer-data-au-lambdabank.cabal [hunk 2]
```diff
@@ -40,6 +41,7 @@ library
                      , wai == 3.2.*
                      , warp == 3.2.*
   hs-source-dirs:      src
+  ghc-options:         -Wall -Werror
```
Comment (benkolera) [inline consumer-data-au-lambdabank/consumer-data-au-lambdabank.cabal]: @ajmcmiddlin Was this off just by accident? I can remove this if it was intentional.

Comment (ajmcmiddlin) [reply consumer-data-au-lambdabank/consumer-data-au-lambdabank.cabal]: Accident. I started with an existing `cabal` file from one of the other packages, so surprised it got lost. Sorry!

**Pull request merged**